### PR TITLE
Update Runtime to v0.13.0

### PR DESCRIPTION
**Pull request by mbaijal**
Update Runtime to v0.13.0 and add the primary identifier field for both ScalableTarget and ScalingPolicy as required. 

### Testing
Integration Tests are passing.
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/scalable_target/sdk.go [hunk 1]
```diff
@@ -120,10 +131,18 @@ func (rm *resourceManager) sdkFind(
...
 	return &resource{ko}, nil
 }
 
+// requiredFieldsMissingFromReadManyInput returns true if there are any fields
+// for the ReadMany Input shape that are required but not present in the
+// resource's Spec or Status
+func (rm *resourceManager) requiredFieldsMissingFromReadManyInput(
+	r *resource,
+) bool {
+	return false
```
[the remark this answers is not in the document]
Comment (mbaijal) [reply pkg/resource/scalable_target/sdk.go:143]: Outside the scope of this PR imo

[the remark this answers is not in the document]
Comment (mbaijal) [reply pkg/resource/scalable_target/sdk.go:143]: > We should be able to use this to replace [this custom logic ](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/blob/main/pkg/resource/scalable_target/custom_api.go#L21)now

But outside the scope of this PR

File: pkg/resource/scaling_policy/delta.go [hunk 2]
```diff
@@ -98,15 +105,19 @@ func newResourceDelta(
...
 		delta.Add("Spec.TargetTrackingScalingPolicyConfiguration", a.ko.Spec.TargetTrackingScalingPolicyConfiguration, b.ko.Spec.TargetTrackingScalingPolicyConfiguration)
 	} else if a.ko.Spec.TargetTrackingScalingPolicyConfiguration != nil && b.ko.Spec.TargetTrackingScalingPolicyConfiguration != nil {
 		if ackcompare.HasNilDifference(a.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification, b.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification) {
 			delta.Add("Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification", a.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification, b.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification)
 		} else if a.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification != nil && b.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification != nil {
-
+			if !reflect.DeepEqual(a.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification.Dimensions, b.ko.Spec.TargetTrackingScalingPolicyConfiguration.CustomizedMetricSpecification.Dimensions) {
```
[the remark this answers is not in the document]
Comment (mbaijal) [reply pkg/resource/scaling_policy/delta.go:118]: I checked the controller logs for any detected diffs but didn't see any based on existing tests.

File: pkg/resource/scalable_target/sdk.go [hunk 3]
```diff
@@ -45,20 +46,30 @@ var (
...
-
-	resp, respErr := rm.sdkapi.DescribeScalableTargetsWithContext(ctx, input)
-	rm.metrics.RecordAPICall("READ_MANY", "DescribeScalableTargets", respErr)
-	if respErr != nil {
-		if awsErr, ok := ackerr.AWSError(respErr); ok && awsErr.Code() == "UNKNOWN" {
+	var resp *svcsdk.DescribeScalableTargetsOutput
+	resp, err = rm.sdkapi.DescribeScalableTargetsWithContext(ctx, input)
+	rm.metrics.RecordAPICall("READ_MANY", "DescribeScalableTargets", err)
+	if err != nil {
+		if awsErr, ok := ackerr.AWSError(err); ok && awsErr.Code() == "UNKNOWN" {
```
[the remark this answers is not in the document]
Comment (mbaijal) [reply pkg/resource/scalable_target/sdk.go:69]: Yes ScalableTarget and so is ScalablePolicy is unique per resource

File: helm/values.yaml [hunk 4]
```diff
@@ -3,8 +3,8 @@
 # Declare variables to be passed into your templates.
 
 image:
-  repository: public.ecr.aws/aws-controllers-k8s/controller
-  tag: applicationautoscaling-v0.0.1
+  repository: public.ecr.aws/aws-controllers-k8s/applicationautoscaling-controller
+  tag: v0.0.1-master
```
[the remark this answers is not in the document]
Comment (mbaijal) [reply helm/values.yaml]: Issue Resolved offline.

File: pkg/resource/scalable_target/sdk.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (surajkota) [reply pkg/resource/scalable_target/sdk.go:69]: Ok for now then

File: pkg/resource/scalable_target/sdk.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (surajkota) [reply pkg/resource/scalable_target/sdk.go:143]: sample for future reference
https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/generator.yaml#L15

File: pkg/resource/scalable_target/resource.go [hunk 5]
```diff
@@ -74,17 +74,37 @@ func (r *resource) Conditions() []*ackv1alpha1.Condition {
...
 // SetIdentifiers sets the Spec or Status field that is referenced as the unique
 // resource identifier
 func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
-	if identifier.NameOrID == nil {
-		return ackerrors.MissingNameIdentifier
+	if r.ko.Status.ACKResourceMetadata == nil {
+		r.ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
 	}
-	r.ko.Spec.ResourceID = identifier.NameOrID
+	r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
```
[the remark this answers is not in the document]
Comment (RedbackThomson) [reply pkg/resource/scalable_target/resource.go:98]: `ResourceID` is the value of `primary_identifier_field_name` - I am recommending `ResourceIds`. The field needs to match the member of the describe input shape.

[the remark this answers is not in the document]
Comment (surajkota) [reply pkg/resource/scalable_target/resource.go:98]: @mbaijal can you make this change so we can merge this PR?

File: pkg/testutil/test_suite_runner.go [hunk 6]
```diff
@@ -69,26 +69,26 @@ func (runner *TestSuiteRunner) RunTests() {
...
+			if r := recover(); r != nil {
+				fmt.Println(RecoverPanicString, r)
+				t.Fail()
+			}
+		}()
+		fmt.Printf("Running test scenario: %s\n", scenario.Name)
+		fixtureCxt := runner.setupFixtureContext(&scenario.Fixture)
+		runner.runTestScenario(t, scenario.Name, fixtureCxt, scenario.UnitUnderTest, &scenario.Expect)
+	})
 }
```
Comment (A-Hilaly) [inline pkg/testutil/test_suite_runner.go:92]: Thanks for the `gofmt`!

File: pkg/resource/scalable_target/manager_test_suite_test.go [hunk 7]
```diff
@@ -17,38 +17,38 @@ import (
...
+	"github.com/ghodss/yaml"
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"go.uber.org/zap/zapcore"
 	"path/filepath"
-	"testing"
 	ctrlrtzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
-	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
-	"go.uber.org/zap/zapcore"
+	"testing"
```
Comment (A-Hilaly) [inline pkg/resource/scalable_target/manager_test_suite_test.go:32]: Needs `goimports` with probably some manual reorganization

File: pkg/resource/scaling_policy/manager_test_suite_test.go [hunk 8]
```diff
@@ -17,38 +17,38 @@ import (
...
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"go.uber.org/zap/zapcore"
 	"path/filepath"
-	"testing"
 	ctrlrtzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
-	ackmetrics "github.com/aws-controllers-k8s/runtime/pkg/metrics"
-	"go.uber.org/zap/zapcore"
+	"testing"
 )
```
Comment (A-Hilaly) [inline pkg/resource/scaling_policy/manager_test_suite_test.go:33]: Ditto

**Review by A-Hilaly — COMMENTED** [review]
LGTM 👍 
Left few comments

File: pkg/resource/scalable_target/manager_test_suite_test.go (hunk 7 above)
Comment (mbaijal) [reply pkg/resource/scalable_target/manager_test_suite_test.go:32]: Amine, we will address this in a separate PR.

**Review by surajkota — APPROVED** [review]
/approve

**Pull request merged**